### ISSUE TITLE
[Table, a11y] Arrow key navigation for tables

### ISSUE THREAD

**Issue by cwgormanIBM** [carbon-design-system/carbon#4981]
It looks like arrow key navigation in tables is not present in the examples?

I would expect tables to follow the arrow key navigation pattern in the [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices/#grid). They have [live demos here](https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html).

## Title line template: [Title]: Brief description

-->

## What package(s) are you using?

- [ ] `carbon-components`
- [ ] `carbon-components-react`

## Detailed description

Describe in detail the issue you're having. Is this a feature request (new
component, new icon), a bug, or a general issue?

> Is this issue related to a specific component?

> What did you expect to happen? What happened instead? What would you like to
> see changed?

> What browser are you working in?

> What version of the Carbon Design System are you using?

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1.  Step one
2.  Step two
3.  Step three
4.  etc.

## Additional information

- Screenshots or code
- Notes

## Add labels

Please choose the appropriate label(s) from our existing label list to ensure
that your issue is properly categorized. This will help us to better understand
and address your issue.

**Comment by asudoh** [comment]
Hi 👋 thank you for reporting - As far as I see our data table is closer to https://www.w3.org/TR/wai-aria-practices/examples/table/table.html rather than data grid one, which means that we rely on screen reader's navigation keyboard shortcut to read the content through.

**Comment by cwgormanIBM** [comment]
It looks like that table is fairly simple and just shows some data. Is that pattern still OK if the table rows have checkboxes, buttons, overflow menus, etc?

**Comment by joshblack** [comment]
@cwgormanIBM yeah, it seems fair that this functionality be added for data grids but may not be appropriate for tables. When we start adding in interactive elements, like overflow menus, it seems like we should shift into the data grid territory where arrow key navigation comes into play.

What do you think @dakahn ?

**Comment by dakahn** [comment]
@joshblack totally agree! A data grid seems like a more fitting set of guidelines to follow with this component. That's a larger undertaking given we've been building this thing to the data table spec for a few years now, but this is a great suggestion.

**Comment by dakahn** [comment]
I'm gonna close this one up. As a general rule giving keyboard users element's to focus that don't really do anything is confusing and can be frustrating. That said, WCAG "data grids" do explicitly do this.

Perhaps this could be a feature enhancement request (along with the presumed breaking changes flipping this from a table to grid may cause) though in the future. Into continuing the discussion. I'll open the ticket and ref here

**Comment by cwgormanIBM** [comment]
What is the recommended keyboard interaction when a table contains interactive elements such as links, buttons, expandable rows, or overflow menus?
